packages/ui-kit/src/components/PieChart/PieChart.tsx

```diff
@@ -0,0 +1,326 @@
+import { useCounter, useLeaderboard } from '../../hooks'
+import { ErrorFallback } from '../ErrorFallback'
+import { Loader } from '../Loader'
+import { withContainer } from '../withContainer'
+import { PieChartProps, PieChartdData } from './PieChart.types'
+
+const defaultChartColorPlatte: string[] = [
+  componentStyles.color_blue_800,
+  componentStyles.color_blue_700,
+  componentStyles.color_blue_600,
+  componentStyles.color_blue_500,
+  componentStyles.color_blue_400,
+  componentStyles.color_blue_300,
+  componentStyles.color_blue_200,
+  componentStyles.color_blue_100,
+  componentStyles.color_blue_50,
+  componentStyles.color_blue_25
+]
+
+let idCounter = 0
+
+export const PieChartComponent = React.forwardRef<HTMLDivElement, PieChartProps>(
+  (
+    {
+      variant = 'pie',
+      headers,
+      rows,
+      query,
+      error,
+      loaderProps,
+      errorFallbackProps,
+      card = false,
+      className,
+      style,
+      baseTheme = 'lightTheme',
+      loading: isLoadingStatic = false,
+      chartProps = {},
+      chartConfigProps,
+      ...rest
+    }: PieChartProps,
+    forwardedRef: any
+  ) => {
+    const innerRef = React.useRef<HTMLDivElement>(null)
+    const { componentContainer, setRef } = useCombinedRefsCallback({ innerRef, forwardedRef })
+    const { theme, chartConfig } = useSetupTheme<'pie' | 'doughnut'>({ componentContainer, baseTheme })
+    const [propsMismatch, setPropsMismatch] = React.useState(false)
+
+    const idRef = React.useRef(idCounter++)
+    const id = `piechart-${idRef.current}`
+
+    /**
+     * The html node where the chart will render
+     */
+    const canvasRef = React.useRef<HTMLCanvasElement>(null)
+    const chartRef = React.useRef<ChartJS | null>()
+
+    /**
+     * Checks if the component is in `static` or `connected` mode
+     */
+    const isStatic = !query
+
+    /**
+     * Fetches the leaderboard data from the API
+     */
+    const {
+      data: leaderboardData,
+      isLoading: leaderboardIsLoading,
+      error: leaderboardHasError
+    } = useLeaderboard({ ...query })
+
+    /**
+     * Fetches the counter data from the API
+     */
+    const { data: counterData, isLoading: counterIsLoading, error: counterHasError } = useCounter({ ...query })
+
+    const isLoading = leaderboardIsLoading || counterIsLoading
+
+    const hasError = leaderboardHasError || counterHasError
+
+    const isPie = variant === 'pie'
+    const isDoughnut = variant === 'doughnut'
+
+    const renderChart = React.useCallback(
+      (data?: PieChartdData) => {
+        if (!canvasRef.current || !data || !theme || !chartConfig) {
+          return
+        }
+
+        const { chartColorPlatte = defaultChartColorPlatte } = chartProps
+
+        const labels = (data.rows?.map((row) => row.slice(0, row.length - 1)) as PieChartLabels) ?? []
+
+        const values =
+          data.rows?.map((row) => (row[row.length - 1] === null ? null : Number(row[row.length - 1]))) || []
```

Comment:
According to this we are enabling `null` values that could be returned by the api to be passed into the `values`, what would happen if chartjs gets a null value in the array? Would that break? Would that not show at all? What would happen to the label if it fails silently?

If this doesn't break the chart then probably we should log a warning in this case wdyt?